### PR TITLE
CSS-7027 global async queries & readme update

### PR DESCRIPTION
**Pull request by AmberCharitos**
This PR addresses the issue of chart and dashboard timeout's experienced by the GE team on long running queries by enabling asynchronous querying for charts and dashboards. For concurrent queries the `WEB_SERVER_TIMEOUT` limit is enforced at 60 seconds. Enabling asynchronous querying avoids these timeouts and allows a large number of users/queries to be executed from charts or dashboards asynchronously. 

Even if we end up not enabling this for our production/staging environments because we want to encourage teams to optimise their queries, this will be a useful addition to the charm for other organisations/individuals.

Additionally there is an update to the `README` discussing the feature flags in general and then this flag specifically.

### CODE REVIEW

File: src/structured_config.py [hunk 1]
```diff
@@ -152,6 +155,27 @@ def sqlalchemy_max_overflow_validator(cls, value: str) -> Optional[int]:
...
+            value: global-async-query-polling-delay value
+
+        Returns:
+            int_value: integer for global-async-query-polling-delay configuration
+
+        Raises:
+            ValueError: in the case when the value is out of range
+        """
+        int_value = int(value)
+        if 500 <= int_value <= 5000:
```
Comment (maabujayyab) [inline src/structured_config.py:175]: We're better off with literals (min and max values here) being defined somewhere that shows what they mean and can be tweaked if needed.

**Review by maabujayyab — APPROVED** [review]
LGTM with a small note

**Review by mertalpt — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -192,6 +192,46 @@ juju scale-application superset-k8s-worker 3
...
+
+Note: `GLOBAL_ASYNC_QUERIES` by default is not enabled as this requires at least 1 worker deployment.
+
+Additional feature flags are available which are not currently implemented by the Superset Charm, a complete list can be found [here](https://github.com/apache/superset/blob/master/RESOURCES/FEATURE_FLAGS.md), with descriptions [here](https://preset.io/blog/feature-flags-in-apache-superset-and-preset/).
+
+## Global asynchronous queries (disabled by default)
+The `GLOBAL_ASYNC_QUERIES` feature flag enables asynchronous querying for Superset charts and dashboards (otherwise only available with SQLlabs). Queries are added to the Celery queue where they are picked up by the next available Superset worker, while the worker executes the queries the server receives updates via regular HTTP polling.
+
+With asynchronous queries Superset can avoid browser timeouts that occur when executing long-running queries and instead default to the database timeouts as the limitation. Additionally this allows the system to handle a large number of concurrent users and queries without siginicant impact on performance, preventing bottlenecks during peak periods.
```
Comment (kelkawi-a) [inline README.md:212]: significant*

File: config.yaml [hunk 3]
```diff
@@ -152,6 +152,21 @@ options:
...
+  global-async-queries:
+    description: |
+      Enable asynchronous querying for charts and dashboard.
+    type: boolean
+    default: False
+  global-async-queries-jwt:
+    description: |
+      jwt for communication with redis database for async querying
+    type: string
+    default: "18b2f8fcd0d708d270c00508da6e8dfc7a21eff14ea438056809805150439a04"
```
Comment (kelkawi-a) [inline config.yaml:164]: Where is this value from?

**Review by kelkawi-a — APPROVED** [review]
LGTM. You should probably start considering breaking down your `README` into a separate documentation folder following the diataxis framework soon.

File: src/structured_config.py (hunk 1 above)
Comment (AmberCharitos) [reply src/structured_config.py:175]: I have followed this pattern for all of the values in `structured_config` - I'll make a separate PR which creates literals with definitions for all of them.

File: config.yaml (hunk 3 above)
Comment (AmberCharitos) [reply config.yaml:164]: It's just a default created with `openssl`

**Pull request merged**